### PR TITLE
[sui examples] Adds "publisher" and "display" pages

### PR DESCRIPTION
**Pull request by damirka**
Sui Examples update - adds pages for new features: `package::Publisher` and `display`.

### CODE REVIEW

File: doc/book/src/basics/display.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+## Description
+
+Sui Object Display is a template engine which allows for on-chain display configuration for type to be handled off-chain by the ecosystem. It has the ability to use an object's data for substitution into a template string.
+
+There's no limitation to what fields can be set, all object properties can be accessed via the `{property}` syntax and inserted as a part of the template string (see examples for the illustration).
+
+## Example
+
+For the Hero module below, the Display would vary based on the "name", "id" and "img_url" properties of the type "Hero". The template defined in the init function can be represented as:
```
Comment (RonMilton) [inline doc/book/src/basics/display.md:15]:
```suggestion
For the following Hero module, the Display would vary based on the "name", "id" and "img_url" properties of the type "Hero". The template defined in the init function can be represented as:
```
Ashok would be mad if I didn't call this out...

**Review by RonMilton — COMMENTED** [review]
Content looks good. Not approving so code review isn't missed.

File: doc/book/src/basics/display.md (hunk 1 above)
Comment (ronny-mysten) [inline doc/book/src/basics/display.md:15]:
```suggestion
For the following Hero module, the Display would vary based on the "name", "id" and "img_url" properties of the type "Hero". The template defined in the init function can be represented as:
```
Ashok would be upset if I didn't point this out...

**Review by ronny-mysten — COMMENTED** [review]
Content looks good. Not approving so code review doesn't get missed.

File: doc/book/examples/sources/basics/publisher.move [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    /// a `Publisher` object. Which can be used to set Display
+    /// or manage the transfer policies in the `Kiosk` system.
+    fun init(otw: OWNER, ctx: &mut TxContext) {
+        package::claim_and_keep(otw, ctx)
+    }
+}
+
+/// A module that utilizes the `Publisher` object to give a token
+/// of appreciation and a `TypeOwnerCap` for the owned type.
+module examples::type_owner {
```
Comment (sblackshear) [inline doc/book/examples/sources/basics/publisher.move:28]: I think this example is a bit hard to follow without seeing a usage of the `TypeOwnerCap`.

In general, I think it's most useful for these examples to show how to write "product-y" code (and then show how to use it). "Infra -y" code like this is also great, but is targeting a somewhat smaller/different audience I think.

File: doc/book/examples/sources/basics/publisher.move [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    use sui::object::{Self, UID};
+    use sui::tx_context::TxContext;
+    use sui::package::{Self, Publisher};
+
+    /// Trying to claim ownership of a type with a wrong `Publisher`.
+    const ENotOwner: u64 = 0;
+
+    /// A capability granted to those who want an "objective"
+    /// confirmation of their ownership :)
+    struct TypeOwnerCap<phantom T> has key, store {
```
Comment (sblackshear) [inline doc/book/examples/sources/basics/publisher.move:38]: Would this be a utility type that we would provide, or do we want folks to roll their own?

**Review by sblackshear — APPROVED** [review]
Nice!

File: doc/book/src/basics/publisher.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Publisher
+
+Publisher Object serves as a way to represent the publisher authority. The object itself does not imply any specific use case and has only two main functions: `package::from_module<T>` and `package::from_package<T>` which allow checking whether a type `T` belongs to a module or a package for which the `Publisher` object was created.
+
+We strongly advise to issue the `Publisher` object for most of the packages that define new Objects - it is required to set the "Display" as well as to allow the type to be traded in the "Kiosk" ecosystem.
+
+> Although `Publisher` itself is a utility, it enables the _"proof of ownership"_ functionality, for example, it is crucial for [the Object Display](./display.md).
+
+To set up a Publisher, an One-Time-Witness (OTW) is required - this way we ensure the `Publisher` object is initialized only once for a specific module (but can be multiple for a package) as well as that the creation function is called in the publish transaction.
```
Comment (amnn) [inline doc/book/src/basics/publisher.md:9]: Nit (it's weird but even though `O` is a vowel, it does not have a vowel sound at the beginning of "One").

```suggestion
To set up a Publisher, a One-Time-Witness (OTW) is required - this way we ensure the `Publisher` object is initialized only once for a specific module (but can be multiple for a package) as well as that the creation function is called in the publish transaction.
```

File: doc/book/examples/sources/basics/publisher.move (hunk 2 above)
Comment (amnn) [reply doc/book/examples/sources/basics/publisher.move:28]: I think `Publisher` is the Infra-y (reusable) part here -- @damirka, do you have any interesting product-specific examples (outside of Display) for wanting to know that a caller created the type?

**Review by amnn — APPROVED** [review]
LGTM as well!  Great to see our showcase of examples growing :)

File: doc/book/examples/sources/basics/publisher.move (hunk 3 above)
Comment (damirka) [reply doc/book/examples/sources/basics/publisher.move:38]: It's really hard to come up with a real production-ready case for the Publisher. It unblocks generic modules (like Collectible, Royalty, Kiosk, Display etc) but I don't know a single simple app that would require proof of type ownership.

File: doc/book/examples/sources/basics/publisher.move (hunk 2 above)
Comment (damirka) [reply doc/book/examples/sources/basics/publisher.move:28]: Answered below. There's no simple application for a Publisher as it is a creator-specific feature to show that they do own a type and that they have control over them.
